### PR TITLE
Adding cloud options to scan_parquet

### PR DESCRIPTION
**Pull request by Bidek56**
Adding cloud features
Adding cloud options to scan_parquet
Changing scan_parquet options to match Python Polars

### CODE REVIEW

**Comment by Bidek56** [comment]
I have not added a S3 cloud test, b/c I am getting `polars.exceptions.ComputeError: Generic S3 error: Error after 2 retries in 3.225558459s` and not sure of the correct options to get it working ATM.

File: polars/io.ts [hunk 1]
```diff
@@ -476,33 +476,56 @@ interface RowCount {
 }
 
 interface ScanParquetOptions {
-  columns?: string[] | number[];
-  numRows?: number;
+  nRows?: number;
```
Comment (universalmind303) [inline polars/io.ts:479]: i prefer `numRows` here.

File: polars/io.ts [hunk 2]
```diff
@@ -476,33 +476,56 @@ interface RowCount {
...
+  cache?: boolean;
   parallel?: "auto" | "columns" | "row_groups" | "none";
   rowCount?: RowCount;
-  cache?: boolean;
   rechunk?: boolean;
-  hive_partitioning?: boolean;
+  rowCountName?: string;
+  rowCountOffset?: number;
```
Comment (universalmind303) [inline polars/io.ts:485]: We already have `rowCount` object that users should use. No need to have it here too. 

`scanParquet({rowCount: {name: 'name', offset: 0}, ...})`

**Review by universalmind303 — CHANGES_REQUESTED** [review]

File: polars/io.ts (hunk 1 above)
Comment (Bidek56) [reply polars/io.ts:479]: I changed it to make it consistent with the Rust and Python versions which have `n_rows`

**Pull request merged**